examples/chip-tool/commands/group/Commands.h

```diff
@@ -0,0 +1,372 @@
+                if (FindKeySetId(fabricIndex, group.group_id, keysetId))
+                {
+                    fprintf(stderr, "  | 0x%-12x  0x%-13x  %-50s |\n", group.group_id, keysetId, group.name);
+                }
+                else
+                {
+                    fprintf(stderr, "  | 0x%-12x  %-15s  %-50s |\n", group.group_id, "None", group.name);
+                }
+            }
+            it->Release();
+        }
+        fprintf(stderr, "  +-------------------------------------------------------------------------------------+\n");
+        SetCommandExitStatus(CHIP_NO_ERROR);
+        return CHIP_NO_ERROR;
+    }
+};
+
+class AddGroup : public CHIPCommand
+{
+public:
+    AddGroup(CredentialIssuerCommands * credsIssuerConfig) : CHIPCommand("add-group", credsIssuerConfig)
+    {
+        AddArgument("groupName", &groupName);
+        AddArgument("groupId", chip::kUndefinedGroupId, UINT16_MAX, &groupId);
+    }
+    chip::System::Clock::Timeout GetWaitDuration() const override { return chip::System::Clock::Seconds16(20); }
+
+    CHIP_ERROR RunCommand() override
+    {
+        if (strlen(groupName) > CHIP_CONFIG_MAX_GROUP_NAME_LENGTH || groupId == chip::kUndefinedGroupId)
+        {
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+
+        chip::FabricIndex fabricIndex;
+        CurrentCommissioner().GetFabricIndex(&fabricIndex);
+        chip::Credentials::GroupDataProvider * groupDataProvider = chip::Credentials::GetGroupDataProvider();
+        chip::Credentials::GroupDataProvider::GroupInfo group;
+
+        group.SetName(groupName);
+        group.group_id = groupId;
+        ReturnErrorOnFailure(groupDataProvider->SetGroupInfo(fabricIndex, group));
+
+        SetCommandExitStatus(CHIP_NO_ERROR);
+        return CHIP_NO_ERROR;
+    }
+
+private:
+    char * groupName;
+    chip::GroupId groupId;
+};
+
+class RemoveGroup : public CHIPCommand
+{
+public:
+    RemoveGroup(CredentialIssuerCommands * credsIssuerConfig) : CHIPCommand("remove-group", credsIssuerConfig)
+    {
+        AddArgument("groupId", chip::kUndefinedGroupId, UINT16_MAX, &groupId);
+    }
+    chip::System::Clock::Timeout GetWaitDuration() const override { return chip::System::Clock::Seconds16(5); }
+
+    CHIP_ERROR RunCommand() override
+    {
+        if (groupId == chip::kUndefinedGroupId)
+        {
+            ChipLogError(chipTool, "Invalid group Id : 0x%x", groupId);
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+
+        chip::FabricIndex fabricIndex;
+        CurrentCommissioner().GetFabricIndex(&fabricIndex);
+        chip::Credentials::GroupDataProvider * groupDataProvider = chip::Credentials::GetGroupDataProvider();
+        ReturnErrorOnFailure(groupDataProvider->RemoveGroupInfo(fabricIndex, groupId));
+        SetCommandExitStatus(CHIP_NO_ERROR);
+        return CHIP_NO_ERROR;
+    }
+
+private:
+    chip::GroupId groupId;
+};
+
+class ShowKeySet : public CHIPCommand
```

Comment:
Looks like my comment on this was not clear.  I was suggesting changing the class name to match what it does, not changing the command name.  So this:
```suggestion
class ShowKeySets : public CHIPCommand
```

Reply:
done